### DISCUSSION TITLE
Автоматиеское присвоение номера новой таблице в цикле

### DISCUSSION THREAD

**Discussion by Leo-ui-528** [totumonline/totum-ru-issues-and-discussions#249] [Как это сделать?] [answered]
Здравствуйте, подскажите пожалуйста, как сделать автоприсвоение номера в поле номер при создании новой таблице в цикле?

**Comment by erpic-ru** [comment, marked as answer]
Обсуждали [здесь](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/51)

**Reply by ttm-alexeypolunin** [reply]
Угу, делать это нужно в таблице **Циклов**, а внутрь цикла уже брать этот номер через функцию `select` при добавлении.

**Comment by Leo-ui-528** [comment]
Получилось, спасибо

**Comment by Leo-ui-528** [comment]
Почти, в таблице циклов работает(select(table: $#ntn; field: 'numb'; order: 'numb' desc) + 1) при добавлении, а select при добавлении в цикле select(table: 'cycle'; field: 'numb';  order: 'numb' asc)при добавлении. всегда пишет один, подскажите как это исправить?

**Reply by ttm-alexeypolunin** [reply]
В приведенном коде так написано: select(table: 'cycle'; field: 'numb'; order: 'numb' **asc**)

`asc` означает брать по возрастанию, а это `1` тк он самый ранний.

**Comment by Leo-ui-528** [comment]
Номера присваиваются при добавлении нового договора - цикла, но при открытии порядковый номер меняется на последний, подскажите пожалуйста как решить это? код в поле: 
= : str`$row2 +"/"+ $row1`
row1: select(table: 'novaya_tablica'; field: 'num'; order: 'id' desc)
row2: select(table: 'napravleniya_deyatelnosti'; field: 'kod'; where: 'id' = #f_napravlenie_deyatelnosti)
![image](https://user-images.githubusercontent.com/75322878/212324904-6618b9b6-1835-4dca-8650-aa0a785174e2.png)

**Reply by ttm-alexeypolunin** [reply]
Может речь идет про [«Исполнять код только при добавлении»](https://ru.docs.totum.online/fields#fields-settings-codeOnlyInAdd)

**Comment by Leo-ui-528** [comment]
Да но, тогда не работает склеивание при выборе кода направления деятельности 
row2: select(table: 'napravleniya_deyatelnosti'; field: 'kod'; where: 'id' = #f_napravlenie_deyatelnosti)

**Reply by Leo-ui-528** [reply]
![image](https://user-images.githubusercontent.com/75322878/212611338-feb7b8ea-e9a1-480f-b788-37c4bb77aa2c.png)

**Comment by Leo-ui-528** [comment]
Разбил по 2 ячейкам, так работает как надо

**Comment by Leo-ui-528** [comment]
Подскажите пожалуйста, как лучше и каким кодом можно изменять нумерацию с 1 по истечении года?

**Reply by erpic-ru** [reply]
https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/103
